controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java

```diff
@@ -43,36 +39,36 @@ public void throwIfUnauthorizedForDeploy(Principal principal,
                                              Environment environment,
                                              Tenant tenant,
                                              ApplicationId applicationId) {
-        if (athenzCredentialsRequired(environment, tenant, applicationId, principal))
-            checkAthenzCredentials(principal, tenant, applicationId);
-    }
-
-    // TODO: inline when deployment via ssh is removed
-    private boolean athenzCredentialsRequired(Environment environment, Tenant tenant, ApplicationId applicationId, Principal principal) {
-        if (!environmentRequiresAuthorization(environment))  return false;
-
-        if (! isScrewdriverPrincipal(principal))
-            throw loggedForbiddenException(
-                    "Principal '%s' is not a screwdriver principal, and does not have deploy access to application '%s'",
-                    principal.getName(), applicationId.toShortString());
+        if (!environmentRequiresAuthorization(environment))  return;
 
-        return tenant.isAthensTenant();
-    }
+        if (principal == null) {
+            throw loggedUnauthorizedException("Principal '%s' is not authenticated.", principal.getName());
```

Comment:
Won't `principal.getName()` throw a NPE here? Consider adding test that hits this path.

Reply:
Will fix.